### PR TITLE
move connectivity to experimental

### PR DESCRIPTION
**Pull request by bjlittle**
### Description
This PR migrates the `Connectivity` and `ConnectivityMetadata` classes (ref: #3968) to `iris.experimental.ugrid`.

Note that, this is targeting the `mesh-data-model` feature branch.

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

File: lib/iris/coords.py [hunk 1]
```diff
@@ -625,7 +624,7 @@ def xml_element(self, doc):
         # otherwise.
         if isinstance(self, Coord):
             values_term = "points"
-        elif isinstance(self, Connectivity):
+        elif hasattr(self, "indices"):
```
Comment (bjlittle) [inline lib/iris/coords.py:627]: Ideally, we'd want to change this back, but only once `iris.experimental.ugrid` is pulled in to the core of `iris`.

File: lib/iris/common/metadata.py [hunk 2]
```diff
@@ -1577,46 +1456,42 @@ def values(self):
...
-    ConnectivityMetadata.__eq__,
-    ConnectivityMetadata.equal,
     CoordMetadata.__eq__,
     CoordMetadata.equal,
     CubeMetadata.__eq__,
     CubeMetadata.equal,
     DimCoordMetadata.__eq__,
     DimCoordMetadata.equal,
-)
+]
```
Comment (bjlittle) [inline lib/iris/common/metadata.py:1494]: Ideally, we'd want to change these back to `tuple`s once `iris.experimental.ugrid` is merged into the core of `iris`.

File: lib/iris/common/metadata.py [hunk 3]
```diff
@@ -1577,46 +1456,42 @@ def values(self):
 
 
 #: Convenience collection of lenient metadata combine services.
-SERVICES_COMBINE = (
+SERVICES_COMBINE = [
```
Comment (trexfeathers) [inline lib/iris/common/metadata.py:1459]:
```suggestion
# TODO: change lists back to tuples once CellMeasureMetadata is re-integrated
# here (currently in experimental.ugrid).
SERVICES_COMBINE = [
```

File: lib/iris/experimental/ugrid.py [hunk 4]
```diff
@@ -5,42 +5,632 @@
...
+
+
+__all__ = [
+    "Connectivity",
+    "ConnectivityMetadata",
+]
+
+
+# Configure the logger.
+logger = get_logger(__name__, fmt="[%(cls)s.%(funcName)s]")
```
Comment (trexfeathers) [inline lib/iris/experimental/ugrid.py:39]: I'm not clear why this is necessary? `logger` isn't used here. Perhaps it's used by the parent classes, but in that case I assume we wouldn't need to import it here too?

**Review by trexfeathers — CHANGES_REQUESTED** [review]
Thanks @bjlittle! A few minor changes/clarifications please 🙂

File: lib/iris/experimental/ugrid.py (hunk 4 above)
Comment (bjlittle) [reply lib/iris/experimental/ugrid.py:39]: It's interesting that `flake8` doesn't complain about this... I though that it might.

Happy to remove it :+1:

**Comment by trexfeathers** [comment]
I really like this concept, since it makes it possible to release some of these features before too long. There may be some Iris issues that have gone on the back burner for a while, but we can show the world that we're still working hard!

**Review by trexfeathers — APPROVED** [review]

**Comment by bjlittle** [comment]
@trexfeathers Awesome, thanks :+1:

**Pull request merged**